### ISSUE TITLE
install issues on linux

### ISSUE THREAD

**Issue by robitalec** [ropensci/codemetar#224]
I'm having trouble installing `codemetar`. I have previously had it installed and it worked without any issues. Today, I tried to use it and it crashed my RStudio session (the bomb) once I ran `library(codemetar)` or `codemetar::`. Every time I retried, crash+bomb. 

(I have updated all my packages, R version, removed the package, lock files, tried with `devtools::install_github` and `install.packages`). For what it's worth, the devtools approach returns the same error. 

##

```r
install.packages('codemetar')
```

```
Installing package into ‘/home/alecr/R/x86_64-pc-linux-gnu-library/3.5’
(as ‘lib’ is unspecified)
trying URL 'https://cran.rstudio.com/src/contrib/codemetar_0.1.6.tar.gz'
Content type 'application/x-gzip' length 293171 bytes (286 KB)
==================================================
downloaded 286 KB

* installing *source* package ‘codemetar’ ...
** package ‘codemetar’ successfully unpacked and MD5 sums checked
** R
** inst
** byte-compile and prepare package for lazy loading


#
# Fatal error in ../src/platform-linux.cc, line 776
# CHECK_EQ(0, result) failed
#   Expected: 0
#   Found: 22
#

/usr/lib64/R/bin/INSTALL: line 34: 27222 Done                    echo 'tools:::.install_packages()'
     27223 Trace/breakpoint trap   (core dumped) | R_DEFAULT_PACKAGES= LC_COLLATE=C "${R_HOME}/bin/R" $myArgs --slave --args ${args}
Warning in install.packages :
  installation of package ‘codemetar’ had non-zero exit status

The downloaded source packages are in
	‘/tmp/RtmpAJKOZg/downloaded_packages’
```

## 

```r
sessionInfo()
```
```
R version 3.5.2 (2018-12-20)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Arch Linux

Matrix products: default
BLAS: /usr/lib/libopenblasp-r0.3.4.so
LAPACK: /usr/lib/liblapack.so.3.8.0

locale:
 [1] LC_CTYPE=en_CA.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_CA.UTF-8        LC_COLLATE=en_CA.UTF-8    
 [5] LC_MONETARY=en_CA.UTF-8    LC_MESSAGES=en_CA.UTF-8   
 [7] LC_PAPER=en_CA.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_CA.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] magrittr_1.5  usethis_1.4.0

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.0        rstudioapi_0.9.0  devtools_2.0.1    pkgload_1.0.2    
 [5] R6_2.3.0          rlang_0.3.1       tools_3.5.2       pkgbuild_1.0.2   
 [9] sessioninfo_1.1.1 cli_1.0.1         withr_2.1.2       remotes_2.0.2    
[13] yaml_2.2.0        assertthat_0.2.0  digest_0.6.18     rprojroot_1.3-2  
[17] crayon_1.3.4      processx_3.2.1    callr_3.1.1       fs_1.2.6         
[21] ps_1.3.0          curl_3.3          testthat_2.0.1    memoise_1.1.0    
[25] glue_1.3.0        compiler_3.5.2    desc_1.2.0        backports_1.1.3  
[29] prettyunits_1.0.2
```

**Comment by cboettig** [comment]
weird, no idea.  No problems here on Debian Linux:

```

**Comment by cboettig** [comment]
Can you paste the full log?  ` ../src/platform-linux.cc, line 776` doesn't refer to a `codemetar` file, but not sue what it is, seems related to byte-compiling when installing the package?  

(also added my session info above in edit, posted prematurely with the wrong hotkey)

**Comment by robitalec** [comment]
What/where is the full log?

This issue (https://github.com/ropensci/jsonvalidate/issues/12) may be related... 

If we look at the first level of dependencies for `codemetar`:
```r
dep1 <- tools::package_dependencies('codemetar')
dep1
```

```
$codemetar
 [1] "jsonlite"    "jsonld"     
 [3] "git2r"       "devtools"   
 [5] "memoise"     "methods"    
 [7] "stats"       "stringi"    
 [9] "readr"       "desc"       
[11] "usethis"     "whisker"    
[13] "tibble"      "crul"       
[15] "gh"          "stringr"    
[17] "sessioninfo" "purrr"      
[19] "curl"       
```

and second level dependencies:

```r
dep2 <- lapply(dep1, function(x) tools::package_dependencies(x))
dep2
```

```r
$codemetar
$codemetar$jsonlite
[1] "methods"

$codemetar$jsonld
[1] "V8"       "jsonlite" "curl"    

$codemetar$git2r
[1] "graphics" "utils"   

(...)
```

We see that `V8` is a dependency... So it looks like a `V8` / Arch issue, which is why it would work fine for Debian. I think we can close it as unrelated for `codemetar`...

**Comment by cboettig** [comment]
yes, that does indeed look  related. Can you try installing the dev branch from GitHub instead:

```r
# install.packages("remotes")
remotes::install_github("ropensci/codmetar@dev")
```

The packages that introduce a dependency on V8 (jsonld) have been moved to Suggests and are not required for most of the `codemetar` functionality.  If V8 on Arch really is this issue, as it probably is, this should sidestep the problem.  

cc @maelle -- are we ready for another PR from `dev` into `master` to address this or are we waiting on anything further?

**Comment by robitalec** [comment]
Nice that did it. 
(Note: typo above in the remotes command, should be `remotes::install_github("ropensci/codemetar@dev")`.

Thanks @cboettig!

**Comment by cboettig** [comment]
nice, thanks!  (typo corrected, good catch).
